dali/python/nvidia/dali/plugin/pytorch/__init__.py

```diff
@@ -795,3 +853,266 @@ def __next__(self) -> List[Dict[str, torch.Tensor]]:
     DENSE_TAG: str = "dense"
     SPARSE_LIST_TAG: str = "sparse_list"
     SPARSE_COO_TAG: str = "sparse_coo"
+
+
+class DALIProxy:
+    """
+    Proxy to communicate to send processing requests to a DALI pipeline running on the main loop.
+    This is used by PyTorch data workers to assign some processing to the loaded samples, which can
+    execute in the main process via the GPU.
+
+    Background: As the PyTorch workers run on separate processes, using the GPU directly from each
+    of those is not ideal for performance, due to the usage of several CUDA contexts.
+    """
+
+    def __init__(self, input_names):
+        """
+        Initializes a new DALI proxy instance.
+
+        Args:
+            input_names (list): list of strings representing the inputs to the pipeline. Those should match
+                                the names of the ``external_source`` nodes in the DALI pipeline.
+        """
+        self.input_names = input_names
+        self.num_inputs = len(input_names)
+        # Multi-process queue used to transfer data from the pytorch workers to the main process
+        self.send_q = mp.Queue()
+        # Multi-process queue used by the main process to remember the actual order of execution of the requests
+        self.order_q = mp.Queue()
```

Comment:
It would be useful to state in this comment which process writes to that queue and which reads from it.

Reply:
Wonderful, thank you!

Reply:
Closing as outdated